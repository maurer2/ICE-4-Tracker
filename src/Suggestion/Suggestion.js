import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './Suggestion.module.css';

const Suggestion = (props) => {
  const { selectedTrainNumber, trainNumber } = props;

  return (
    <li
      className={ classNames(
        styles.suggestion,
        {
          [styles['suggestion--is-active']]: selectedTrainNumber === trainNumber
        },
      )}
    >
      { trainNumber }
    </li>
  );
};

export default Suggestion;

Suggestion.propTypes = {
  selectedTrainNumber: PropTypes.string,
  trainNumber: PropTypes.string,
};
