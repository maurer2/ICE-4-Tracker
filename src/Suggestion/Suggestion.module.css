.suggestion {
  padding: 0.5rem;
}

.suggestion--is-active {
  background: #6b6b6b;
  color: #fff;
}

.button {
  display: block;
  border: 0;
  text-align: left;
  background: none;
  color: inherit;
}
