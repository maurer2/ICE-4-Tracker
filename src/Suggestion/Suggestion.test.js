import React from 'react';
import Suggestion from './Suggestion';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('Suggestion', () => {
  const mockedHandleClick = jest.fn();

  const wrapper = shallow(
    <Suggestion
      isActive={true}
      trainNumber={'500'}
      cbHandleClickEvent={mockedHandleClick}
    />
  );

  it('Suggestion should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
