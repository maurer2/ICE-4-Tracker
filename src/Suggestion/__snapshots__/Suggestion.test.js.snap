// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestion Suggestion should match snapshot 1`] = `
<li
  className="suggestion suggestion--is-active"
  onClick={[Function]}
>
  500
</li>
`;
